.github/workflows/codeql.yml

```diff
@@ -0,0 +1,42 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  schedule:
+    - cron: "1 18 * * 0"
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ c, python ]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: ${{ matrix.language }}
+          queries: +security-and-quality
```

Comment:
```suggestion
          queries: +security-and-quality
          query-filters:
           - exclude:
                      id: cpp/bad-strncpy-size
                      id: cpp/certificate-not-checked
                      id: cpp/cgi-xss
                      id: cpp/cleartext-storage-buffer
                      id: cpp/cleartext-storage-file
                      id: cpp/command-line-injection
                      id: cpp/external-entity-expansion
                      id: cpp/incorrect-allocation-error-handling
                      id: cpp/insufficient-key-size
                      id: cpp/missing-check-scanf
                      id: cpp/non-https-url
                      id: cpp/path-injection
                      id: cpp/pointer-overflow-check
                      id: cpp/potential-system-data-exposure
                      id: cpp/resource-not-released-in-destructor
                      id: cpp/return-stack-allocated-memory
                      id: cpp/signed-overflow-check
                      id: cpp/sql-injection
                      id: cpp/system-data-exposure
                      id: cpp/tainted-format-string
                      id: cpp/tainted-format-string-through-global
                      id: cpp/tainted-permissions-check
                      id: cpp/unbounded-write
                      id: cpp/uncontrolled-allocation-size
                      id: cpp/uncontrolled-arithmetic
                      id: cpp/uncontrolled-process-operation
                      id: cpp/unsafe-strncat
                      id: cpp/unsafe-use-of-this
                      id: cpp/user-controlled-bypass
                      id: cpp/using-expired-stack-address
```

Reply:
This list of analyses that seemed to have taken the longest amount of time. We should review the wisdom of disabling them.

[ ] Does this suggested commit reduce the time taken by 2hrs (to 30 minutes)
[ ] Can exclude only a subset of this list and achieve the same performance results?
[ ] Evaluate this list for things we should not exclude.
[ ] Look for other ways of reducing the run time.

Reply:
We could also potentially have a separate configurations for pushes and PRs. The PR configuration could include the cheap checks which is hopefully style related stuff, and the push could check for everything else that we could add afterwards.